source/extensions/resource_monitors/fixed_heap/fixed_heap_monitor.cc

```diff
@@ -22,10 +25,25 @@ FixedHeapMonitor::FixedHeapMonitor(
 }
 
 void FixedHeapMonitor::updateResourceUsage(Server::ResourceUpdateCallbacks& callbacks) {
-  const size_t physical = stats_->reservedHeapBytes();
-  const size_t unmapped = stats_->unmappedHeapBytes();
-  ASSERT(physical >= unmapped);
-  const size_t used = physical - unmapped;
+
+  size_t used;
```

Comment:
imo, would be cleaner to have this be initialized the necessary value and you can then make it const.

You could achieve by putting the stuff below into a lambda